nav2_bringup/spawn_turtlebot3/src/spawn_turtlebot3/spawn_turtlebot.py

```diff
@@ -0,0 +1,77 @@
+"""
+spawn_turtlebot.py
+
+Script used to spawn a turtlebot in a generic position
+"""
+import os
+import sys
+import rclpy
+from ament_index_python.packages import get_package_share_directory
+from gazebo_msgs.srv import SpawnEntity
+import xml.etree.ElementTree as ET
+
+
+def main():
+    """ Main for spwaning turtlebot node """
+    # Get input arguments from user
+    argv = sys.argv[1:]
+
+    # Start node
+    rclpy.init()
+    node = rclpy.create_node("entity_spawner")
+
+    node.get_logger().info(
+        'Creating Service client to connect to `/spawn_entity`')
+    client = node.create_client(SpawnEntity, "/spawn_entity")
+
+    node.get_logger().info("Connecting to `/spawn_entity` service...")
+    if not client.service_is_ready():
+        client.wait_for_service()
+        node.get_logger().info("...connected!")
+
+    # Get path to the turtlebot3 burgerbot
+    sdf_file_path = os.path.join(
+        get_package_share_directory("turtlebot3_gazebo"), "models",
+        "turtlebot3_burger", "model.sdf")
+
+    # We need to remap the transform (/tf) topic so each robot has its own.
+    # We do this by adding `ROS argument entries` to the sdf file for
+    # each plugin broadcasting a transform. These argument entries provide the
+    # remapping rule, i.e. /tf -> /<robot_id>/tf
+    tree = ET.parse(sdf_file_path)
+    root = tree.getroot()
+    for plugin in root.iter('plugin'):
+        if 'turtlebot3_diff_drive' in plugin.attrib.values():
+            # The only plugin we care for now is 'diff_drive' which is
+            # broadcasting a transform between`odom` and `base_footprint`
+            break
+
+    ros_params = plugin.find('ros')
+    ros_tf_remap = ET.SubElement(ros_params, 'argument')
+    ros_tf_remap.text = '/tf:=/' + argv[1] + '/tf'
```

Comment:
If user does not provide `robot_id` we should be able to generate a unique id.  Also if the user provides an id that is already taken, we should either generate a unique id or send an error msg that says a robot with this id is already spawn.

Reply:
I think the generic name of "robot" is sufficient if none provided

Reply:
If a robot with the name `robot` is already spawned, this could either cause the system to crash or it would overwrite the first robot with the same name.  I think each spawned robot should have a unique id.

Reply:
I think the case of someone trying to launch overwriting robots is a good reason to simply crash than assign other IDs

Reply:
I am thinking of cases when someone is trying to spawn a new robot from a terminal and by accident the user provides the same id. I think we shouldn't just let it to crash, we can either ignore the new spawn and provide a feedback to the user or spawn a new robot with a unique ID.

Reply:
I think part of the issue with spawning with a new/random id/uuid is that they might have different parameters and if they've messed up a launch file / shell script to bringup a system with multiple robots, its good to know right then and there its wrong vs masking it and wondering why robot `N` isnt acting as expected. This is especially true if when launching `robot_n` all the services / actions are namespaced in it (which must be true) `robot_n/navigate_to_pose` then you have no idea what the topic to operate the robot would even be since its then just assigned

Reply:
I'd prefer that it doesn't crash but just reject the spawn request and show an error message.

Reply:
That would be fine as well

Reply:
Just push a change, we now use default values for name, namespace, and position (`robot`, `robot`, `0, 0, 0`). If a robot already exists with the same name it will not be spawned.